packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -489,50 +489,46 @@ class _AdaptiveScaffoldState extends State<AdaptiveScaffold> {
           bodyOrientation: widget.bodyOrientation,
           bodyRatio: widget.bodyRatio,
           internalAnimations: widget.internalAnimations,
-          primaryNavigation: widget.destinations != null &&
-                  widget.selectedIndex != null
-              ? SlotLayout(
-                  config: <Breakpoint, SlotLayoutConfig>{
-                    widget.mediumBreakpoint: SlotLayout.from(
-                      key: const Key('primaryNavigation'),
-                      builder: (_) => AdaptiveScaffold.standardNavigationRail(
-                        width: widget.navigationRailWidth,
-                        selectedIndex: widget.selectedIndex,
-                        destinations: widget.destinations
-                            .map((_) => AdaptiveScaffold.toRailDestination(_))
-                            .toList(),
-                        onDestinationSelected: widget.onSelectedIndexChange,
-                      ),
-                    ),
-                    widget.largeBreakpoint: SlotLayout.from(
-                      key: const Key('primaryNavigation1'),
-                      builder: (_) => AdaptiveScaffold.standardNavigationRail(
-                        width: widget.extendedNavigationRailWidth,
-                        extended: true,
-                        selectedIndex: widget.selectedIndex,
-                        destinations: widget.destinations
-                            .map((_) => AdaptiveScaffold.toRailDestination(_))
-                            .toList(),
-                        onDestinationSelected: widget.onSelectedIndexChange,
-                      ),
-                    ),
-                  },
-                )
-              : null,
-          bottomNavigation: widget.destinations != null &&
```

Comment:
Since `widget.destinations` can never be null it is always true (caught by `core_lints`). `true && (conditions)` can be simplified to just `(conditions)`